transport/x/grpc/handler.go

```diff
@@ -62,6 +62,21 @@ func (h *handler) handle(
 	decodeFunc func(interface{}) error,
 	interceptor grpc.UnaryServerInterceptor,
 ) (interface{}, error) {
+	responseMD := metadata.New(nil)
```

Comment:
I almost committed that actually, but left this as is for now because I want to do a followup PR that addresses the lack of pooling buffers in transport/x/grpc.